driver-core/src/main/com/mongodb/client/model/expressions/MapExpression.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import static com.mongodb.client.model.expressions.Expressions.of;
+
+public interface MapExpression<T extends Expression> extends Expression {
```

Comment:
Right you are - `getMap` is missing, and needs to be added. This would be `doc.getMap("instock").entrySet()`. Passing test:

        DocumentExpression doc = of(Document.parse("{ instock: { warehouse1: 2500, warehouse2: 500 } }"));
        assertExpression(
                Arrays.asList(
                        Document.parse("{'k': 'warehouse1', 'v': 2500}"),
                        Document.parse("{'k': 'warehouse2', 'v': 500}")),
                doc.getMap("instock").entrySet(),
                "{'$objectToArray': {'$getField': {'input': {'$literal': "
                        + "{'instock': {'warehouse1': 2500, 'warehouse2': 500}}}, 'field': 'instock'}}}");

Let's get #1050 and #1052 resolved, and I will be able to merge those and #1053 in, and then rebase to fix this.

Reply:
Added getMap and isMapOr in [aae819e](https://github.com/mongodb/mongo-java-driver/pull/1054/commits/aae819e96f8cade8c7411da341e10571e35a9423).

Incidentally, the fact that maps are equal to documents introduces an unavoidable way of converting between them (using isMapOr/isDocumentOr).